internal/hardware/validator.go

```diff
@@ -63,31 +68,60 @@ func isNvme(name string) bool {
 	return strings.HasPrefix(name, "nvme")
 }
 
-func ListValidDisks(inventory *models.Inventory, minSizeRequiredInBytes int64) []*models.Disk {
-	var disks []*models.Disk
-	for _, disk := range inventory.Disks {
-		if disk.SizeBytes >= minSizeRequiredInBytes && funk.ContainsString([]string{"HDD", "SSD"}, disk.DriveType) {
-			disks = append(disks, disk)
-		}
+// DiskIsEligible checks if a disk is relevant for installation by testing
+// it against a list of predicates. Also returns all the reasons the disk
+// was found to be not eligible
+func (v *validator) DiskIsEligible(disk *models.Disk) (eligible bool, notEligibleReasons []string) {
+	if minSizeBytes := gbToBytes(v.MinDiskSizeGb); disk.SizeBytes < minSizeBytes {
+		notEligibleReasons = append(notEligibleReasons,
+			fmt.Sprintf(
+				"Disk is too small (disk only has %s, but %s are required)",
+				humanize.Bytes(uint64(disk.SizeBytes)), humanize.Bytes(uint64(minSizeBytes))))
+	}
+
+	if allowedDriveTypes := []string{"HDD", "SSD"}; !funk.ContainsString(allowedDriveTypes, disk.DriveType) {
+		notEligibleReasons = append(notEligibleReasons,
+			fmt.Sprintf("Drive type is %s, it must be one of %s.",
+				disk.DriveType, strings.Join(allowedDriveTypes, ", ")))
 	}
 
+	eligible = len(notEligibleReasons) == 0
+
+	return
```

Comment:
I don't know what the general opinion on this is, but I don't like utilizing named return values like this.
I would prefer local vars and an explicit return.

Reply:
I personally prefer it. @filanov, what's your take on it?